loopy/isl_helpers.py

```diff
@@ -574,7 +574,17 @@ def find_max_of_pwaff_with_params(pw_aff, n_allowed_params):
 # }}}
 
 
-# {{{ subst_into_pwqpolynomial
+# {{{ subst_into_pw(qpolynomial|aff)
+
+def set_dim_name(obj, dt, pos, name):
+    assert isinstance(name, str)
+    if isinstance(obj, isl.PwQPolynomial):
+        return obj.set_dim_name(dt, pos, name)
+    elif isinstance(obj, isl.PwAff):
+        return obj.set_dim_id(dt, pos, isl.Id.read_from_str(obj.get_ctx(), name))
```

Comment:
Yikes! What makes this necessary?

Reply:
```python
In [1]: import islpy as isl

In [2]: isl.Id("foo")
Out[2]: Id("foo@0x5643f0b47d60")

In [3]: isl.Id.read_from_str(isl.DEFAULT_CONTEXT, "foo")
Out[3]: Id("foo")

In [4]: hex(id(None))
Out[4]: '0x5643f0b47d60'
```

`isl.Id` attaches `None` as the user object which leads to a lot of problems when performing binary operations between 2 isl objects.

Reply:
Not sure I understand. Why not simply use `set_dim_name`?

Reply:
ISL doesn't have isl_pw_aff_set_dim_name. It does have isl_pw_aff_get_dim_name, isl_pw_aff_find_dim_by_name, etc. Not quite sure why they missed it (should it be reported as a bug?).

Reply:
```
$ nm --dynamic /usr/lib/x86_64-linux-gnu/libisl.so | grep isl_pw_aff_set
0000000000075ed0 T isl_pw_aff_set_dim_id
0000000000073cf0 T isl_pw_aff_set_dim_name
000000000007c110 T isl_pw_aff_set_rational
0000000000075dc0 T isl_pw_aff_set_tuple_id
```
But it's not in the header... I think it's worth sending Sven a patch. (And add a comment here to say what it is you're working around.)

Reply:
(If he resists, we could conceivably paper over this in islpy.)